src/cockpit/transports.py

```diff
@@ -219,7 +219,13 @@ def _create_write_queue(self, data: bytes) -> None:
         self._protocol.pause_writing()
 
     def write(self, data: bytes) -> None:
-        assert not self._closing
+        # this is a race condition with subprocesses: if we get and process the the "exited"
+        # event before seeing BrokenPipeError, we'll try to write to a closed pipe.
+        # Do what the standard library does and ignore, instead of assert
+        if self._closing:
+            logger.warning('ignoring write() to closing transport fd %i', self._out_fd)
```

Comment:
This is possibly too loud for something that can occur due to no error at all in a race condition on old Python versions....

Reply:
Right -- now that we understand it, I lowered it to debug().